Spatial.gd

```diff
@@ -1137,27 +1137,70 @@ func _on_MarkersUI_item_edited():
 	apply_category_visibility_to_nodes(category_item)
 
 
-func _on_ImportPath_pressed():
-	$Control/Dialogs/ImportPackDialog.show()
+func _on_ImportTaco_pressed():
+	$Control/Dialogs/ImportPackDialogs/ImportTacoPackDialog.show()
 
+func _on_ImportBurrito_pressed():
+	$Control/Dialogs/ImportPackDialogs/ImportBurritoPackDialog.show()
 
-func _on_ImportPackDialog_dir_selected(dir):
-	var user_data_dir = str(OS.get_user_data_dir())
-	var args: PoolStringArray = [
-		"--input-taco-path", dir,
+enum MarkerPackType {
+	XML,
+	GUILDPOINT,
+}
+
+func _on_ImportBurritoPackDialog_dir_selected(dir: String):
+	test_name(dir, MarkerPackType.GUILDPOINT)
+
+func _on_ImportTacoPackDialog_dir_selected(dir: String):
+	test_name(dir, MarkerPackType.XML)
+
+func test_name(dir: String, type):
+	var args: PoolStringArray
+	if type == MarkerPackType.XML:
+		args.push_back("--input-taco-path")
+	elif type == MarkerPackType.GUILDPOINT:
+		args.push_back("--input-guildpoint-path")
+	args.push_back(dir)
+
+	args.append_array([
 		"--input-guildpoint-path", ProjectSettings.globalize_path(self.saved_markers_dir),
 		"--output-guildpoint-path", ProjectSettings.globalize_path(self.saved_markers_dir),
-		"--output-split-guildpoint-path", ProjectSettings.globalize_path(self.unsaved_markers_dir)
-	]
-	FileHandler.call_xml_converter(args)
-	save_hashes()
-	load_guildpoint_markers(self.map_id)
+		"--split-by-category",
+		"--output-guildpoint-path", ProjectSettings.globalize_path(self.unsaved_markers_dir),
+		"--split-by-map-id",
+	])
+
+	var duplicate_categories: Dictionary = FileHandler.call_xml_converter(args)
+	if duplicate_categories.empty():
+		save_hashes()
+		load_guildpoint_markers(self.map_id)
+		return
 
+	var overwrite_confirmation: ConfirmationDialog = $Control/Dialogs/ImportPackDialogs/OverwriteConfirm
+	overwrite_confirmation.dialog_text = "The following marker packs will be overwritten. \n" + PoolStringArray(duplicate_categories.keys()).join("\n")
+	overwrite_confirmation.popup_centered()
+
+	# Await result of pop up
+	var return_val = overwrite_confirmation.wait_for_confirmation()
+	if typeof(return_val) == TYPE_OBJECT and return_val.is_class("GDScriptFunctionState"):
+		return_val = yield(return_val, "completed")
+	if return_val == true:
+		for category in duplicate_categories.keys():
+			for file in duplicate_categories[category]:
+				if file.begins_with(ProjectSettings.globalize_path(self.saved_markers_dir)):
+					FileHandler.delete_file(file)
+		duplicate_categories = FileHandler.call_xml_converter(args)
+		if duplicate_categories.empty():
+			save_hashes()
+			load_guildpoint_markers(self.map_id)
+		else:
+			print("Duplicates could not be resolved.")
```

Comment:
I think there is a function that is something like `push_error` native to godot.
Also toast.

Reply:
Yes `push_error` will add it to the debugger. This is a good place to add it.